### DISCUSSION TITLE
Group Interrupt (GINT) Enable at DTS file - LPC55xxx

### DISCUSSION THREAD

**Discussion by Aravindh1910** [zephyrproject-rtos/zephyr#64322] [Q&A] [answered]
Hi Everyone,

currently I'm developing a zephyr based project on LPC5536 Microcontroller.
where I have to setup 10 GPIO interrupts.
But, as per user manual, there is limitation that upto 8 GPIO pin interrupts (PINT) are allowed to configure.
So I decided to configure the remaining or all pins using Group interrupts (GINT).

As far I see in the example boards (lpxpresso55xxx board dts file,)  I am able to configure pin interrupts (PINT),

```
gpio0: gpio@0 {
		compatible = "nxp,lpc-gpio";
		reg = <0x8c000 0x278c>;
		interrupts = <4 2>,<5 2>,<6 2>,<7 2>;             //PINT (0 to 3)
		gpio-controller;
		#gpio-cells = <2>;
		port = <0>;
};
gpio1: gpio@1 {
		compatible = "nxp,lpc-gpio";
		reg = <0x8c000 0x278c>;
		interrupts = <32 2>,<33 2>,<34 2>,<35 2>;      //PINT (4 to 7)
		gpio-controller;
		#gpio-cells = <2>;
		port = <1>;
};
```

In similar way, I need support/suggestions to configure group interrupts (GINT) in dts file.

Thanks in advance.!!!

**Comment by DerekSnell** [comment, marked as answer]
Hi @Aravindh1910 ,
Yes, this is a limitation with the current LPC GPIO driver.  The GINT and port interrupts are not supported yet.  Today only the PINT is supported, which limits GPIO interrupts to 8 pins.

One option to work around this is to use the MCUXpresso SDK APIs in the HAL to configure the GINT, and register the ISR with the Zephyr kernel.  The MCUXpresso SDK has this [GINT example](https://github.com/nxp-mcuxpresso/mcux-sdk-examples/tree/main/lpcxpresso55s36/driver_examples/gint).

Best regards

**Reply by Aravindh1910** [reply]
Hi @DerekSnell ,

Thank you for your suggestion.
